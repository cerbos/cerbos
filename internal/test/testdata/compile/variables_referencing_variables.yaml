# yaml-language-server: $schema=../.jsonschema/CompileTestCase.schema.json
---
mainDef: resource_policies/example.yaml
inputDefs:
  resource_policies/example.yaml:
    apiVersion: api.cerbos.dev/v1
    resourcePolicy:
      version: default
      resource: example
      variables:
        local:
          one: "1"
          two: "2"
          three: variables.one + variables.two
          four: V.one + V.three
