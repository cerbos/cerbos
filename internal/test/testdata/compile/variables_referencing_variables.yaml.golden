{
  "fqn": "cerbos.resource.example.vdefault",
  "resourcePolicy": {
    "meta": {
      "fqn": "cerbos.resource.example.vdefault",
      "resource": "example",
      "version": "default"
    },
    "policies": [
      {
        "orderedVariables": [
          {
            "name": "one",
            "expr": {
              "original": "1",
              "checked": {
                "typeMap": {
                  "1": {
                    "primitive": "INT64"
                  }
                },
                "sourceInfo": {
                  "location": "<input>",
                  "lineOffsets": [
                    2
                  ],
                  "positions": {
                    "1": 0
                  }
                },
                "expr": {
                  "id": "1",
                  "constExpr": {
                    "int64Value": "1"
                  }
                }
              }
            }
          },
          {
            "name": "two",
            "expr": {
              "original": "2",
              "checked": {
                "typeMap": {
                  "1": {
                    "primitive": "INT64"
                  }
                },
                "sourceInfo": {
                  "location": "<input>",
                  "lineOffsets": [
                    2
                  ],
                  "positions": {
                    "1": 0
                  }
                },
                "expr": {
                  "id": "1",
                  "constExpr": {
                    "int64Value": "2"
                  }
                }
              }
            }
          },
          {
            "name": "three",
            "expr": {
              "original": "variables.one + variables.two",
              "checked": {
                "referenceMap": {
                  "1": {
                    "name": "variables"
                  },
                  "3": {
                    "overloadId": [
                      "add_bytes",
                      "add_double",
                      "add_duration_duration",
                      "add_duration_timestamp",
                      "add_timestamp_duration",
                      "add_int64",
                      "add_list",
                      "add_string",
                      "add_uint64"
                    ]
                  },
                  "4": {
                    "name": "variables"
                  }
                },
                "typeMap": {
                  "1": {
                    "mapType": {
                      "keyType": {
                        "primitive": "STRING"
                      },
                      "valueType": {
                        "dyn": {}
                      }
                    }
                  },
                  "2": {
                    "dyn": {}
                  },
                  "3": {
                    "dyn": {}
                  },
                  "4": {
                    "mapType": {
                      "keyType": {
                        "primitive": "STRING"
                      },
                      "valueType": {
                        "dyn": {}
                      }
                    }
                  },
                  "5": {
                    "dyn": {}
                  }
                },
                "sourceInfo": {
                  "location": "<input>",
                  "lineOffsets": [
                    30
                  ],
                  "positions": {
                    "1": 0,
                    "2": 9,
                    "3": 14,
                    "4": 16,
                    "5": 25
                  }
                },
                "expr": {
                  "id": "3",
                  "callExpr": {
                    "function": "_+_",
                    "args": [
                      {
                        "id": "2",
                        "selectExpr": {
                          "operand": {
                            "id": "1",
                            "identExpr": {
                              "name": "variables"
                            }
                          },
                          "field": "one"
                        }
                      },
                      {
                        "id": "5",
                        "selectExpr": {
                          "operand": {
                            "id": "4",
                            "identExpr": {
                              "name": "variables"
                            }
                          },
                          "field": "two"
                        }
                      }
                    ]
                  }
                }
              }
            }
          },
          {
            "name": "four",
            "expr": {
              "original": "V.one + V.three",
              "checked": {
                "referenceMap": {
                  "1": {
                    "name": "V"
                  },
                  "3": {
                    "overloadId": [
                      "add_bytes",
                      "add_double",
                      "add_duration_duration",
                      "add_duration_timestamp",
                      "add_timestamp_duration",
                      "add_int64",
                      "add_list",
                      "add_string",
                      "add_uint64"
                    ]
                  },
                  "4": {
                    "name": "V"
                  }
                },
                "typeMap": {
                  "1": {
                    "mapType": {
                      "keyType": {
                        "primitive": "STRING"
                      },
                      "valueType": {
                        "dyn": {}
                      }
                    }
                  },
                  "2": {
                    "dyn": {}
                  },
                  "3": {
                    "dyn": {}
                  },
                  "4": {
                    "mapType": {
                      "keyType": {
                        "primitive": "STRING"
                      },
                      "valueType": {
                        "dyn": {}
                      }
                    }
                  },
                  "5": {
                    "dyn": {}
                  }
                },
                "sourceInfo": {
                  "location": "<input>",
                  "lineOffsets": [
                    16
                  ],
                  "positions": {
                    "1": 0,
                    "2": 1,
                    "3": 6,
                    "4": 8,
                    "5": 9
                  }
                },
                "expr": {
                  "id": "3",
                  "callExpr": {
                    "function": "_+_",
                    "args": [
                      {
                        "id": "2",
                        "selectExpr": {
                          "operand": {
                            "id": "1",
                            "identExpr": {
                              "name": "V"
                            }
                          },
                          "field": "one"
                        }
                      },
                      {
                        "id": "5",
                        "selectExpr": {
                          "operand": {
                            "id": "4",
                            "identExpr": {
                              "name": "V"
                            }
                          },
                          "field": "three"
                        }
                      }
                    ]
                  }
                }
              }
            }
          }
        ]
      }
    ]
  }
}
